### ISSUE TITLE
Early stopping and validation metrics logging during training T2I 

### ISSUE THREAD

**Issue by segalinc** [huggingface/diffusers#5364]
Any plan to add a early stopping approach and loggin validation metrics like FID and IS when training text to image model ?

**Comment by segalinc** [comment]
What I mean is, sometimes when you train ML models, instead of letting them train for the max number of steps, during the validation you compute some metrics, let's say accuracy for a classification problem. Over each epoch you compute this metrics and if after a while you don't see any improvement or it starts decreasing you want to stop training the model so it won't overfit and continue training.
In the case of SD I was thinking there could be a similar way where given the validation images you can compute FID or IS or LPIPS score, keep track of them and interrupt the training if these don't improve for some consecutive epochs.
